presto-spark-classloader-interface/src/main/java/com/facebook/presto/spark/classloader_interface/PrestoSparkNativeExecutionShuffleManager.java

```diff
@@ -71,19 +73,27 @@ private static <T> T instantiateClass(String className, SparkConf conf)
         }
     }
 
+    protected void registerShuffleHandle(BaseShuffleHandle handle, int mapId)
+    {
+        partitionIdToShuffleHandle.put(mapId, handle);
+        shuffleIdToBaseShuffleHandle.put(handle.shuffleId(), handle);
+    }
+
     @Override
     public <K, V, C> ShuffleHandle registerShuffle(int shuffleId, int numMaps, ShuffleDependency<K, V, C> dependency)
     {
-        shuffleNumMaps.put(shuffleId, numMaps);
         return fallbackShuffleManager.registerShuffle(shuffleId, numMaps, dependency);
     }
 
     @Override
     public <K, V> ShuffleWriter<K, V> getWriter(ShuffleHandle handle, int mapId, TaskContext context)
     {
-        shuffleDependencyMap.put(mapId, handle);
-        int shuffleId = handle.shuffleId();
-        return new EmptyShuffleWriter<>(getNumOfPartitions(shuffleId));
+        if (!(handle instanceof BaseShuffleHandle)) {
+            throw new RuntimeException("Only support ShuffleHandle inheriting from BaseShuffleHandle");
```

Comment:
Why RuntimeException? If it is a internal constrain, use

requireNonNull(handle, "handle is null");
checkState (handle instanceof BaseShuffleHandle, "class %s is not instance of BaseShuffleHandle", handle.getClass().getName());